GitCommands/Git/GitModule.cs

```diff
@@ -39,6 +39,16 @@ public sealed class GitModule : IGitModule
         private readonly IRevisionDiffProvider _revisionDiffProvider = new RevisionDiffProvider();
         private readonly IGitCommandRunner _gitCommandRunner;
         private readonly IExecutable _gitExecutable;
+        private readonly Lazy<Encoding> _filesEncoding;
+
+        public GitModule(GitModule module) : this(module.WorkingDir)
+        {
+            _gitCommonDirectory = module._gitCommonDirectory;
+            if (module._filesEncoding.IsValueCreated)
+            {
+                _filesEncoding = new Lazy<Encoding>(() => module._filesEncoding.Value);
```

Comment:
Any reason we can't write `_filesEncoding = module._filesEncoding;`?

Reply:
Because you cannot assign Lazy fields directly

Reply:
Func<T> could be used instead

Reply:
This specific situation changed previously, also changed Lazy handling